driver/src/orderbook/streamed/state.rs

```diff
@@ -646,24 +647,54 @@ mod tests {
             ..Default::default()
         };
         state = state.apply_event(&Event::Trade(event), 1).unwrap();
+        let event = SolutionSubmission {
+            submitter: address(4),
+            burnt_fees: 23.into(),
+            ..Default::default()
+        };
+        state = state
+            .apply_event(&Event::SolutionSubmission(event), 1)
+            .unwrap();
 
-        let event = Trade {
-            owner: address(2),
+        let account_state_ = account_state(&state, 2);
+        assert_eq!(account_state_.read_balance(0, address(2)), U256::from(12));
+        assert_eq!(account_state_.read_balance(1, address(2)), U256::from(9));
+        assert_eq!(account_state_.read_balance(0, address(3)), U256::from(8));
+        assert_eq!(account_state_.read_balance(1, address(3)), U256::from(11));
+        assert_eq!(account_state_.read_balance(0, address(4)), U256::from(23));
+        assert_eq!(
+            state
+                .orders
+                .get(&(address(2), 0))
+                .unwrap()
+                .get_used_amount(2),
+            1
+        );
+        assert_eq!(
+            state
+                .orders
+                .get(&(address(3), 0))
+                .unwrap()
+                .get_used_amount(2),
+            2
+        );
+
+        let event = TradeReversion {
+            owner: address(3),
             order_id: 0,
-            executed_sell_amount: 4,
-            executed_buy_amount: 3,
+            executed_sell_amount: 2,
+            executed_buy_amount: 1,
             ..Default::default()
         };
-        state = state.apply_event(&Event::Trade(event), 1).unwrap();
+        state = state.apply_event(&Event::TradeReversion(event), 1).unwrap();
```

Comment:
This seems like a strange thing to change. Was there something wrong with the test before?

Reply:
Before the test had trade reversions but no second solution submission which doesn't make sense. So I changed the test to revert and submit a new solution which increases coverage because it tests that the fee from the previous solution gets reverted properly.